### PR TITLE
[23.1] Fix ordering of data libraries from API

### PR DESCRIPTION
**Pull request by martenson**
plus a race condition and visual indicator/default sort

fixes the issue @jennaj reported

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/managers/libraries.py [hunk 1]
```diff
@@ -208,6 +210,8 @@ def list(self, trans, deleted: Optional[bool] = False) -> Tuple[Query, Dict[str,
                 prefetched_ids["allowed_library_add_ids"] = allowed_library_add_ids
                 prefetched_ids["allowed_library_modify_ids"] = allowed_library_modify_ids
                 prefetched_ids["allowed_library_manage_ids"] = allowed_library_manage_ids
+        sort_column = getattr(Library, "name")
+        query = query.order_by(asc(func.lower(sort_column)))
```
Comment (jdavcs) [inline lib/galaxy/managers/libraries.py]: why not just `Library.name` instead of `sort_column`?

Comment (martenson) [reply lib/galaxy/managers/libraries.py]: I did that in a subsequent commit when linter yelled at me. I used to have something else there but there is actually no more options for this endpoint.

**Comment by mvdbeek** [comment]
The selenium error looks relevant: https://github.com/galaxyproject/galaxy/actions/runs/5350220192/jobs/9702546472?pr=16300#step:11:5125

**Comment by martenson** [comment]
@mvdbeek yeah it was, it expected unsorted initial render, which I think is wrong

all tests here would also fail for me locally because of weird admin handling, so I adjusted that to standard (?) here: https://github.com/galaxyproject/galaxy/pull/16300/commits/aaeeb085ae58d27c5ab760b5dfa531ce0d22fb48

**Comment by mvdbeek** [comment]
Thanks @martenson!

**Pull request merged**